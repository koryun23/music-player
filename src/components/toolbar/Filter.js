import React from "react";
import "../../css/toolbar/Toolbar.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { useState } from "react";

export default function Filter(props) {

    const [inputWord, setInputWord] = useState("");

    const onInputWordChange = (event) => {
        setInputWord(event.target.value);
        props.onFilter(event.target.value);
    }

    return (
        <form>
            <div className="filter-form">
                <input type="text" 
                       className="filter-input-field" 
                       placeholder="&#xF002; Filter" 
                       style={{fontFamily: "Arial, FontAwesome"}}
                       value={inputWord}
                       onChange={(event) => onInputWordChange(event)}/>
            </div>
        </form>
    );
}