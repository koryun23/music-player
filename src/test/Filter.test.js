import renderer from "react-test-renderer";
import Filter from "../components/toolbar/Filter";

it('renders correctly', () => {
    const tree = renderer.create(<Filter />).toJSON();
    expect(tree).toMatchSnapshot();
})