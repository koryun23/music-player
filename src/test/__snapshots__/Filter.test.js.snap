// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <div
    className="filter-form"
  >
    <input
      className="filter-input-field"
      onChange={[Function]}
      placeholder="ï€‚ Filter"
      style={
        Object {
          "fontFamily": "Arial, FontAwesome",
        }
      }
      type="text"
      value=""
    />
  </div>
</form>
`;
